tests/kdtree.rs

```diff
@@ -306,3 +313,31 @@ fn handles_drops_correctly() {
     // Post-drop check
     assert_eq!(*drop_counter.lock().unwrap(), 4);
 }
+
+#[test]
+fn handles_remove_correctly() {
+    let item1 = ([0f64], 1);
+    let item2 = ([100f64], 2);
+    let item3 = ([45f64], 3);
+    let item4 = ([55f64], 4);
+
+    // Build a kd tree
+    let dimensions = 1;
+    let capacity_per_node = 2;
+    let mut kdtree = KdTree::with_capacity(dimensions, capacity_per_node);
+
+    kdtree.add(&item1.0, item1.1).unwrap();
+    kdtree.add(&item2.0, item2.1).unwrap();
+    kdtree.add(&item3.0, item3.1).unwrap();
+    kdtree.add(&item4.0, item4.1).unwrap();
+
+    kdtree.remove(&&item3.0, &item3.1).unwrap();
```

Comment:
related to one of the above comment, maybe we can add test cases for no match?

Reply:
Added a test, I'll have to rethink how I'm removing items. It does fail for invalid items.